### PR TITLE
Support line location information in KAPT for top level declarations

### PR DESCRIPTION
**Pull request by yigit**
This CL fixes a bug in KAPT's MAP_DIAGNOSTIC_LOCATIONS flag where the
location mapping for top level elements didn't work.

I also needed to changes the Kapt3IntegrationTest to copy input files,
otherwise, it cannot map file paths since the mock project has no root
location.

Fixes: KT-47934
Test: KotlinKapt3IntegrationTests#testErrorLocationMapping

### CODE REVIEW

**Comment by yigit** [comment]
I've run the following tests locally

`./gradlew :kotlin-annotation-processing:test --tests "org.jetbrains.kotlin.kapt3.test.*"`

Also, I'm not sure if there is a better way to run these tests. The existing error location tests were not running processors but instead relying on errors in code which I couldn't reproduce for top level.

File: plugins/kapt3/kapt3-compiler/test/org/jetbrains/kotlin/kapt3/test/AbstractKotlinKapt3IntegrationTest.kt [hunk 1]
```diff
@@ -123,6 +131,24 @@ abstract class AbstractKotlinKapt3IntegrationTest : KotlinKapt3TestBase() {
         }
     }
 
+    override fun loadMultiFiles(files: List<TestFile>) {
+        val project = myEnvironment.project
+        val psiManager = PsiManager.getInstance(project)
+
+        val tmpDir = KtTestUtil.tmpDir("kaptTest")
```
Comment (gavra0) [inline plugins/kapt3/kapt3-compiler/test/org/jetbrains/kotlin/kapt3/test/AbstractKotlinKapt3IntegrationTest.kt:138]: Delete this dir in tearDown()?

**Review by gavra0 — APPROVED** [review]

File: plugins/kapt3/kapt3-compiler/test/org/jetbrains/kotlin/kapt3/test/AbstractKotlinKapt3IntegrationTest.kt (hunk 1 above)
Comment (yigit) [reply plugins/kapt3/kapt3-compiler/test/org/jetbrains/kotlin/kapt3/test/AbstractKotlinKapt3IntegrationTest.kt:138]: I copied this from AbstractKotlinKapt3Test which does not do anything in teardown. Should I update both? (not sure what is the preferred way in these tests)
https://github.com/JetBrains/kotlin/blob/master/plugins/kapt3/kapt3-compiler/test/org/jetbrains/kotlin/kapt3/test/AbstractKotlinKapt3Test.kt#L77

Comment (zveznicht) [reply plugins/kapt3/kapt3-compiler/test/org/jetbrains/kotlin/kapt3/test/AbstractKotlinKapt3IntegrationTest.kt:138]: I looked at how temp dir is used in other tests and it doesn't seem that it get deleted often. Ideally you should delete it ofc, but I wouldn't fix it everywhere right now.

**Comment by zveznicht** [comment]
Can you please run test Kapt3IT.testLocationMapping. It seems it got broken (or just report line changed). If you couldn't reproduce it locally, I can have a look myself

**Comment by yigit** [comment]
> Can you please run test Kapt3IT.testLocationMapping. It seems it got broken (or just report line changed). If you couldn't reproduce it locally, I can have a look myself

sorry i got busy but i'll try to take a look at it in a few days.

**Comment by yigit** [comment]
> > Can you please run test Kapt3IT.testLocationMapping. It seems it got broken (or just report line changed). If you couldn't reproduce it locally, I can have a look myself
> 
> sorry i got busy but i'll try to take a look at it in a few days.

yes it looks like the wrong line in test (annotation is on line 4 but it is asserting for 3. I assume it was 3 because previously it was defaulting to the container when it cannot match the element itself).

**Comment by yigit** [comment]
done. sorry for the delay.

**Review by Tapchicoma — APPROVED** [review]

**Comment by Tapchicoma** [comment]
Merged into `master` via cherry-pick: 95f990adabc326925cd9341de7c8061535e4e98f

**Comment by yigit** [comment]
Thank you!

**Pull request closed without merging**